back/src/Guide/Database/Set.hs

```diff
@@ -0,0 +1,584 @@
+
+-- | Set item name.
+setItemName :: Uid Item -> "name" :! Text -> ExceptT DatabaseError Transaction ()
+setItemName itemId (arg #name -> name) = do
+  let sql = [r|
+        UPDATE items
+        SET name = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, name) (Statement sql encoder decoder False)
+
+-- | Set item link.
+setItemLink :: Uid Item -> "link" :? Url -> ExceptT DatabaseError Transaction ()
+setItemLink itemId (argF #link -> link) = do
+  let sql = [r|
+        UPDATE items
+        SET link = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParamNullable
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, link) (Statement sql encoder decoder False)
+
+-- | Set item hackage.
+--
+-- "hackage" :? Text == Maybe Text with label.
+setItemHackage :: Uid Item -> "hackage" :? Text -> ExceptT DatabaseError Transaction ()
+setItemHackage itemId (argF #hackage -> hackage) = do
+  let sql = [r|
+        UPDATE items
+        SET hackage = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParamNullable
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, hackage) (Statement sql encoder decoder False)
+
+-- | Set item summary.
+setItemSummary :: Uid Item -> "summary" :! Text -> ExceptT DatabaseError Transaction ()
+setItemSummary itemId (arg #summary -> summary) = do
+  let sql = [r|
+        UPDATE items
+        SET summary = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, summary) (Statement sql encoder decoder False)
+
+-- | Set item notes.
+setItemNotes :: Uid Item -> "notes" :! Text -> ExceptT DatabaseError Transaction ()
+setItemNotes itemId (arg #notes -> notes) = do
+  let sql = [r|
+        UPDATE items
+        SET notes = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, notes) (Statement sql encoder decoder False)
+
+-- | Set item ecosystem.
+setItemEcosystem :: Uid Item -> "ecosystem" :! Text -> ExceptT DatabaseError Transaction ()
+setItemEcosystem itemId (arg #ecosystem -> ecosystem) = do
+  let sql = [r|
+        UPDATE items
+        SET ecosystem = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, ecosystem) (Statement sql encoder decoder False)
+
+-- | Set item to be deleted.
+--
+-- When 'deleted' is True, item will marked as deleted (_categoryItemsDeleted)
+-- otherwise item will marked as notDeleted (_categoryItems)
+setItemDeleted :: Uid Item -> "deleted" :! Bool -> ExceptT DatabaseError Transaction ()
+setItemDeleted itemId (arg #deleted -> deleted) = do
+  mCategoryId <- getCategoryIdByItemMaybe itemId
```

Comment:
Let's have a `getCategoryIdByItem` that will throw `ItemNotFound` when necessary. This way you won't have to handle the `Maybe` and thrown errors will be more precise. `ItemOrCategoryNotFound` is a bad error, it doesn't help with debugging much.